### ISSUE TITLE
Add option to ignore custom static class properties?

### ISSUE THREAD

**Issue by emroussel** [edvardchen/eslint-plugin-i18next#8]
[Using `displayName` strings](https://reactjs.org/docs/react-component.html#displayname) in react class components is a common pattern (especially when using higher order components).

```js
const withForm = () => WrappedComponent => {
    const componentName = WrappedComponent.displayName || WrappedComponent.name;

    class Form extends Component {
        static displayName = `withForm(${componentName})`;

        // ...
    }
}
```

However, this plugin flags `displayName` as an error since it uses an unwrapped string. It can be fixed by abstracting it into a constant like this:

```js
const withForm = () => WrappedComponent => {
    const componentName = WrappedComponent.displayName || WrappedComponent.name;

    const DISPLAY_NAME = `withForm(${componentName})`;

    class Form extends Component {
        static displayName = DISPLAY_NAME;

        // ...
    }
}
```

but it would be nice to be able to keep it in its original form for simplicity and cleanliness.

Do you feel like it's a decent/common enough use case to justify adding a new option to ignore a list of custom static class properties? If yes, happy to open a PR to add this if you point me in the right direction.

**Comment by emroussel** [comment]
@edvardchen any reason this was closed? Would love to hear your reasoning if you don't think this is worth adding.

**Comment by edvardchen** [comment]
My mistake. I reopen it

**Comment by edvardchen** [comment]
I agree we should ignore `displayName` attribute. Before that, you can use `ignoreAttribute `

**Comment by edvardchen** [comment]
Hi @emroussel , the plugin doesn't complain about the following code:
```javascript
const withForm = () => WrappedComponent => {
    const componentName = WrappedComponent.displayName || WrappedComponent.name;

    class Form extends Component {
        static displayName = `withForm(${componentName})`;

        // ...
    }
}
```

Can you give more detail or a demo project that reproduces the issue?

**Comment by emroussel** [comment]
@edvardchen my bad, was trying to give an example of a more common use case and didn't realize it wasn't actually replicating the issue. We're also assigning literal strings in some of our components who share names

```js
export class Account extends Component {
    static displayName = 'AccountContainer';

    // ...
}
```

**Comment by edvardchen** [comment]
That would need new option some kind like `ignoreVariable` to approach this case.  I would work on it.

**Comment by emroussel** [comment]
Sounds good, also happy to help and open a PR for it if you prefer

**Comment by edvardchen** [comment]
Great then leave it to you. :)

**Comment by Ridermansb** [comment]
and about functional components ?

```
HomePage.displayName = 'HomePage';
```

There are any way to ignore this as well ?

**Comment by edvardchen** [comment]
@Ridermansb try new version v6.0.0-3. It will be easier to configure.

For you case 

```js
{
  "mode": "all",
  "object-properties": {
    "include": [],
    "exclude": ["displayName"]
  },
```

Or you can try the default behavior which only validate plain text markup in JSX
